src/parquet/file/reader-internal.cc

```diff
@@ -174,13 +175,19 @@ std::unique_ptr<PageReader> SerializedRowGroup::GetColumnPageReader(int i) {
   }
 
   int64_t bytes_to_read = col.meta_data.total_compressed_size;
-  std::shared_ptr<Buffer> buffer = source_->ReadAt(col_start, bytes_to_read);
+  std::unique_ptr<InputStream> stream;
 
-  if (buffer->size() < bytes_to_read) {
-    throw ParquetException("Unable to read column chunk data");
+  if (chunk_size == 0) {
```

Comment:
This is a sort of ugly. Perhaps we should consider IO option configuration more generally. How does parquet-mr handle this?